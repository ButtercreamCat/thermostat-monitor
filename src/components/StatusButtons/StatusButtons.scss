@import "../../sass/abstracts/variables";
@import "../../sass/components/buttons";

.statusButtonsContainer {
  display: flex;
  flex-wrap: wrap;
  margin: 80px 0 50px;
}

@media only screen and (max-width: 768px) {
  .statusButtonsContainer {
    width: 80%;
    height: auto;
    margin: 40px auto 10px;
    justify-content: center;
  }
}
