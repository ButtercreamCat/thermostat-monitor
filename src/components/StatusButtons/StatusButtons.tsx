import React, { Component } from "react";
import "./StatusButtons.scss";

const modeArr = ["Cool", "Heat", "Fan-on", "Auto", "Off"];

interface StatusButtonsProps {
  handleClick: (status: string) => void;
  status: string;
}
class StatusButtons extends Component<StatusButtonsProps> {
  render() {
    const { handleClick, status } = this.props;
    return (
      <div className="statusButtonsContainer">
        {modeArr.map(mode => (
          <button
            key={mode}
            className={`button button-square-withIcon ${status === mode &&
              "button-active"}`}
            onClick={() => {
              handleClick(mode);
            }}
          >
            <span className={`icon-${mode.toLocaleLowerCase()}`}></span>
            <div>{mode}</div>
          </button>
        ))}
      </div>
    );
  }
}

export default StatusButtons;
