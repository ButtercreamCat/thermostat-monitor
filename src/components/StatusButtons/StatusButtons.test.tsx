import React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import StatusButtons from "./StatusButtons";

let wrapper: ShallowWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;
const handleClickSpy = jest.fn();

beforeEach(() => {
  wrapper = shallow(
    <StatusButtons handleClick={handleClickSpy} status="Off" />
  );
});

test("should render StatusButtons correctly", () => {
  expect(wrapper).toMatchSnapshot();
});

test("should button has correct class name", () => {
  expect(
    wrapper
      .find("button")
      .at(4)
      .hasClass("button button-square-withIcon button-active")
  ).toBe(true);
});

test("should call handleClick prop when click the button", () => {
  wrapper
    .find("button")
    .at(0)
    .simulate("click");
  expect(handleClickSpy).toHaveBeenLastCalledWith("Cool");
});
