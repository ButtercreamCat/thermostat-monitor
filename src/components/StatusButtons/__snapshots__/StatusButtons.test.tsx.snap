// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render StatusButtons correctly 1`] = `
<div
  className="statusButtonsContainer"
>
  <button
    className="button button-square-withIcon false"
    key="Cool"
    onClick={[Function]}
  >
    <span
      className="icon-cool"
    />
    <div>
      Cool
    </div>
  </button>
  <button
    className="button button-square-withIcon false"
    key="Heat"
    onClick={[Function]}
  >
    <span
      className="icon-heat"
    />
    <div>
      Heat
    </div>
  </button>
  <button
    className="button button-square-withIcon false"
    key="Fan-on"
    onClick={[Function]}
  >
    <span
      className="icon-fan-on"
    />
    <div>
      Fan-on
    </div>
  </button>
  <button
    className="button button-square-withIcon false"
    key="Auto"
    onClick={[Function]}
  >
    <span
      className="icon-auto"
    />
    <div>
      Auto
    </div>
  </button>
  <button
    className="button button-square-withIcon button-active"
    key="Off"
    onClick={[Function]}
  >
    <span
      className="icon-off"
    />
    <div>
      Off
    </div>
  </button>
</div>
`;
